dist/v3.0.0/common/JOSE-header-schema.json

```diff
@@ -0,0 +1,39 @@
+{
+  "title": "Version 2 of JWT header",
+  "type": "object",
+  "$id": "http://apicentre.paymentsnz.co.nz/schemas/JOSE-header-schema.json",
+  "definitions": {},
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "additionalProperties": false,
+  "required": ["alg", "kid"],
+  "properties": {
+    "alg": {
+      "$id": "/properties/alg",
```

Comment:
I notice some JSON Schema use `#` as a prefix for the $id, and some schemas do not.  

I don't think these ID values are being referenced within this spec anywhere so might not be an issue, but just worth bringing up. Maybe these are intended to be stable ID values for external use? 

If so, they should all be aligned with or without the prefix. 

https://github.com/json-schema-org/json-schema-spec/issues/1059#issuecomment-832872321

> "id": "#/definitions/foo" has been unambiguously invalid since draft-05 and ambiguously invalid forever (not specified as invalid, but clearly not intended use). "id": "/schema/foo#/definitions/foo" is more ambiguous 
 
It seems only in $ref you should use # (or a fragment), and in $id/id you should use an absolute value
 
The other PRs didn't have JSON Schema just OpenAPI specs, so do not have this issue.

Reply:
It may be more nuanced than that.  The base $id must be an absolute, whereas a sub-schema can be relative.  Moreover, it appears that `#` is valid (required) in $id for [location-independent identifiers](https://json-schema.org/draft-07/draft-handrews-json-schema-01#rfc.section.8.2.3), but this should not include solidus `/`:
> Using JSON Pointer fragments requires knowledge of the structure of the schema. When writing schema documents with the intention to provide re-usable schemas, it may be preferable to use a plain name fragment that is not tied to any particular structural location. This allows a subschema to be relocated without requiring JSON Pointer references to be updated.
To specify such a subschema identifier, the "$id" keyword is set to a URI reference with a plain name fragment (not a JSON Pointer fragment). This value MUST begin with the number sign that specifies a fragment ("#"), then a letter ([A-Za-z]), followed by any number of letters, digits ([0-9]), hyphens ("-"), underscores ("_"), colons (":"), or periods (".")

I'll continue investigating and see if I can find out what the best option is here.  Also happy to take any advice your have :)

Reply:
Thinking about this further - I'm not sure we actually need the $id in property definitions?  They don't appear to be referenced anywhere, and the schemas can always be navigated by JSON pointer.

Should we remove the $id from property definitions @fabiancook @rvbabilonia ?

Reply:
Removing them completely would make sense here. The id of those properties would then be the fragment path by default then right... 

+1 for deleting them for just the JSON schema properties, where not referenced.

Reply:
> The id of those properties would then be the fragment path by default then right...

Correct.

For example, modifying the schema example above, it would look similar to this:
```
  "$id": "http://apicentre.paymentsnz.co.nz/schemas/JOSE-header-schema.json",
  "properties": {
    "alg": {
      "type": "string",
      "description": "The type of algorithm used to sign the JWT",
      "examples": ["PS256"],
      "enum": ["ES256", "PS256"]
  }
... etc ..
```
Then you could still reference `alg` above using `"$ref": "#/properties/alg"` within the same schema, or when including in a different schema with
`"$ref": "http://apicentre.paymentsnz.co.nz/schemas/JOSE-header-schema.json#/properties/alg"`

Seems OK?

Reply:
A-OK from me! Looks good and makes sense.

Reply:
Cool - I've pushed updates to remove $ids from properties.

Also updated the auth server metadata $id and renamed the file (just in case we ever add other metadata).